### PR TITLE
fix(dot/state): clean up scheduled changes once a forced change is applied

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

After some investigation around the forced change dispatched by block `#1,491,033` I found that Gossamer handled the forced change but it was not pruning the stored scheduled changes. Every time a new forced change is applied we should ignore past scheduled changes or scheduled changes from forks.

- https://spec.polkadot.network/#sect-finality-forced-changes
- https://github.com/paritytech/substrate/blob/50de15d8740a129db9c18a6698fbd183b00326a2/client/consensus/grandpa/src/authorities.rs#L511

After the fix we could reach a higher number of synced blocks (1.79M)
image

## Tests

```sh
go test -timeout 10m -run ^TestApplyForcedChanges$ github.com/ChainSafe/gossamer/dot/state --tags=integration -v
```

## Issues

- Closes https://github.com/ChainSafe/gossamer/issues/3203

## Primary Reviewer

@timwu20

### CODE REVIEW

**Review by dimartiro — APPROVED** [review]

File: dot/state/grandpa.go [hunk 1]
```diff
@@ -257,9 +257,10 @@ func (s *GrandpaState) ApplyForcedChanges(importedBlockHeader *types.Header) err
 		return fmt.Errorf("cannot set change set id at block")
 	}
 
-	logger.Debugf("Applying authority set forced change at block #%d",
-		forcedChange.announcingHeader.Number)
+	logger.Debugf("Applied authority set forced change: %s", forcedChange)
 
+	s.forcedChanges.pruneAll()
```
Comment (jimjbrettj) [inline dot/state/grandpa.go:262]: Can there be conflicts when applying a forced change and pruning other forced changes? Does it only prune changes that were requested before the current change?

**Review by jimjbrettj — APPROVED** [review]
Looks good! Just a few comments

File: dot/state/grandpa.go (hunk 1 above)
Comment (EclesioMeloJunior) [reply dot/state/grandpa.go:262]: > Does it only prune changes that were requested before the current change

Comment (EclesioMeloJunior) [reply dot/state/grandpa.go:262]: >Does it only prune changes that were requested before the current change?

No, it prunes the entire set, substrate does it as well https://github.com/paritytech/substrate/blob/50de15d8740a129db9c18a6698fbd183b00326a2/client/consensus/grandpa/src/authorities.rs#L511. 
A forced change once applied should change the entire authority set for all forks https://spec.polkadot.network/#sect-finality-forced-changes

Comment (jimjbrettj) [reply dot/state/grandpa.go:262]: awesome thank you

**Pull request merged**